src/fetch-images.ts

```diff
@@ -45,23 +103,46 @@ export type EventTypes = { pageName: string } & (
   | { type: 'sources'; status: 'fetched' }
   | { type: 'images'; status: 'fetching' }
   | { type: 'images'; status: 'fetched' }
+  | {
+      type: 'image'
+      status: 'error'
+      url: string
+      retriesLeft: number
+      error: Error
+    }
 )
 
 export type ImageOptions = {
   /** The file id to fetch images from. Located in the URL of the Figma file. */
   fileId: string
 
+  /** The page name to fetch images from*/
+  pages?: string[]
+
   /** Filter images to fetch. Fetches all images if omitted. */
   filter?: (component: Component) => boolean
 
   /** The event type as it occurs. */
   onEvent?: (event: EventTypes) => void
+
+  /** Time in MS before a image fetch will timeout and retry */
+  fetchTimeout?: number
+
+  /** Maximum number of retries before any individual image fetch will fail */
+  fetchMaxRetries?: number
+
+  /** Does an expensive DFS on all children in the page to attach the parent name */
+  includeParentName?: boolean
```

Comment:
I think we can remove this option and just include this info by default unless you think it will add that much more computation.